client-websocket/src/main/java/com/github/twitch4j/client/websocket/WebsocketConnectionConfig.java

```diff
@@ -0,0 +1,101 @@
+package com.github.twitch4j.client.websocket;
+
+import com.github.twitch4j.common.util.ExponentialBackoffStrategy;
+import com.github.twitch4j.common.util.ThreadUtils;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.experimental.Accessors;
+import org.apache.commons.lang3.RandomStringUtils;
+
+import java.time.Duration;
+import java.util.Objects;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.function.Consumer;
+
+@Accessors(chain = true, fluent = true, prefix = "")
```

Comment:
With `fluent=true`:
```kotlin
WebsocketConnection() {
	it.baseUrl("https://twitch4j.github.io")
	it.wsPingPeriod(1337)
	it.onTextMessage { msg -> println(msg) }
}
```

&nbsp;

With `fluent=false`:
```kotlin
WebsocketConnection() {
	it.baseUrl = "https://twitch4j.github.io"
	it.wsPingPeriod = 1337
	it.onTextMessage = Consumer { msg -> println(msg) }
}
```

&nbsp;

I personally find the latter kotlin syntax to be more idiomatic, but this makes the event handlers feel clunky in java (e.g., `getOnTextMessage`, `setOnTextMessage`), so no change is really needed here. Since end users are unlikely to directly interface with this class, this is more a general comment on how we should approach other specs...

Even more idiomatic would be to remove the `it.` (e.g., so it's just `wsPingPeriod = 1337`) — but this may be a task for the kotlin module to achieve (via extensions functions + lambdas with receiver to create a type-safe builder)

Reply:
I also prefer `fluent=true` because we can improve kotlin usage with the kotlin module later, but we can't do that for java.  

We could add the kotlin extension function in a follow up PR after #516 is merged. This isn't really required for this spec as this is internal, but if we want to provide the improved usage from kotlin i think it's important to do it for all specs (also in the codegen for helix/... later).